### ISSUE TITLE
[Doc]: Install info takes up too much room on new front page

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#21246]
### Documentation Link

https://matplotlib.org/devdocs/index.html

### Problem

The install info at the top takes ~1/3 of the available vertical space on the front page (before scrolling, but taking into account the fact that the "real" version number will not have a git tag).  While important, it is something that most users have to do only once, whereas they (hopefully) will consult the rest of the docs more than once; therefore, it could be made slightly less tall.

### Suggested improvement

The current panels:
![current](https://user-images.githubusercontent.com/1322974/135594990-4026d111-d3cc-443f-9fe0-867f6c6b38b5.png)
without scrolling (and with a "released" version), the page cuts on my screen just after "API reference".

A few options (just some quick hacks, certainly better styling is possible):
![alt1](https://user-images.githubusercontent.com/1322974/135595487-c93120ac-3697-4e98-ab24-b3309023b1d4.png)
![alt2](https://user-images.githubusercontent.com/1322974/135595183-c9efefbd-8dd2-4595-94a9-9c86700c42db.png)
![alt3](https://user-images.githubusercontent.com/1322974/135595187-2e8aa5ab-77cf-4815-840d-eccd1ce1e5ab.png)
On my screen, without scrolling, the page cuts after the link to pyplot.figure with the first and last design, and after the link to pyplot.subplots_mosaic after the middle one.  I think having these immediately visible is nice to have.

<details>
<summary>For reference, the rst sources</summary>
(comment out the unused ones)

```rst
.. code-block:: bash

   # Using conda:
   conda install matplotlib
   # Or using pip:
   pip install matplotlib

.. csv-table::
   :widths: 1 3

   Using `conda <https://docs.continuum.io/anaconda/>`__,``conda install matplotlib``
   Using `pip <https://pypi.org/project/matplotlib>`__,``pip install matplotlib``

.. csv-table::
   :widths: 1 2 1 2

   Using `conda <https://docs.continuum.io/anaconda/>`__:,``conda install matplotlib``,Using `pip <https://pypi.org/project/matplotlib>`__:,``pip install matplotlib``
```
</details>

### Matplotlib Version

master

### Matplotlib documentation version

(local build)

**Comment by jklymak** [comment]
I'd vote for your second one (side-by-side)

**Comment by timhoffm** [comment]
Side by side is better. Is it possible to have the code in the table as code-block highlighting like before? I think it's valuable to have the commands stand out a bit more than just another font and color.

**Comment by anntzer** [comment]
I gave it a quick try with `.. raw:: html` to set up the table and then interspersing `.. code-blocks::` at the strategic place, but the vertical align is messed up.  Probably just needs some more CSS hacking...
